apps/bartender.js

```diff
@@ -5,42 +5,49 @@ function Questions() {
   this.questions = ['Do ye like yer drinks strong?', 'Do ye like it with a salty tang?', 'Are ye a lubber who likes it bitter?', 'Would ye like a bit of sweetness with yer poison?', 'Are ye one for a fruity finish?'];
 }
 
-// Bartender questions
-
-var barQuestions = new Questions();
-
-// Bar ingredients contructor function
+// Bar ingredients constructor function
 
 function Ingredients() {
-  this.strong = ['glug of rum', 'slug of whisky', 'splash of gin'];
-  this.salty = ['olive on a stick', 'salt-dusted rim', 'rasher of bacon'];
-  this.bitter = ['shake of bitters', 'splash of tonic', 'twist of lemon peel'];
-  this.sweet = ['sugar cube', 'spoonful of honey', 'spash of cola'];
-  this.fruity = ['slice of orange', 'dash of cassis', 'cherry on top'];
+  this.ingredients = [['glug of rum', 'slug of whisky', 'splash of gin'],
+  ['olive on a stick', 'salt-dusted rim', 'rasher of bacon'],
+  ['shake of bitters', 'splash of tonic', 'twist of lemon peel'],
+  ['sugar cube', 'spoonful of honey', 'spash of cola'],
+  ['slice of orange', 'dash of cassis', 'cherry on top']];
+  this.randomIngredient = Math.floor((Math.random() * 3) + 0);
 }
 
-// Bar ingredients
-
-var barIngredients = new Ingredients();
-
 // Pantry constructor function
 
 function Pantry(barIngredients) {
-  this.barIngredients = barIngredients;
+  this.barIngredients = new Ingredients();
 }
 
-// Restaurant Pantry for when adding food questions later
-
-var restaurantPantry = new Pantry(barIngredients);
-
 // Customer preferences constructor function
 
 function Preferences() {
   this.preferences = [];
 }
 
+// Bartender constructor function
+
+function Bartender() {
+  var restaurantPantry = new Pantry();
+  this.createDrink = function(preferences) {
+    var drink = [];
+    for (var i = 0; i < preferences.length; i++) {
+      if (preferences[i]) {
```

Comment:
Looks okay but I think I would break this down a little. Maybe something like:
```javascript
var ingredientsGroup = restaurantPantry.barIngredients.ingredients[i];
var randomIngredient = restaurantPantry.barIngredients.randomIngredient;
drink.push(ingredientsGroup[randomIngredient]);
```